v2/datastream-to-spanner/src/test/java/com/google/cloud/teleport/v2/templates/DataStreamToSpannerSimpleIT.java

```diff
@@ -187,6 +190,219 @@ public void migrationTestWithInsertsOnly() {
+        ChainedConditionCheck.builder(
+                List.of(
+                    uploadDataStreamFile(
+                        jobInfo,
+                        TABLE4,
+                        "backfill.jsonl",
+                        "DataStreamToSpannerSimpleIT/mysql-backfill-AllDatatypeColumns.jsonl"),
+                    SpannerRowsCheck.builder(spannerResourceManager, TABLE4)
+                        .setMinRows(2)
+                        .setMaxRows(2)
+                        .build()))
+            .build();
+
+    // Wait for conditions
+    PipelineOperator.Result result =
+        pipelineOperator()
+            .waitForCondition(createConfig(jobInfo, Duration.ofMinutes(8)), conditionCheck);
+
+    // Assert Conditions
+    assertThatResult(result).meetsConditions();
+
+    assertAllDatatypeColumnsTableBackfillContents();
+  }
+
+  @Test
+  public void migrationTestWithAllDatatypeConversionsWithUpdatesAndDeletes() {
+    // Construct a ChainedConditionCheck with 4 stages.
+    // 1. Send initial wave of events
+    // 2. Wait on Spanner to have events
+    // 3. Send second wave of events
+    // 4. Wait on Spanner to merge second wave of events
+    ChainedConditionCheck conditionCheck =
+        ChainedConditionCheck.builder(
+                List.of(
+                    uploadDataStreamFile(
+                        jobInfo,
+                        TABLE4,
+                        "backfill.jsonl",
+                        "DataStreamToSpannerSimpleIT/mysql-backfill-AllDatatypeColumns.jsonl"),
+                    SpannerRowsCheck.builder(spannerResourceManager, TABLE4)
+                        .setMinRows(2)
+                        .setMaxRows(2)
+                        .build(),
+                    uploadDataStreamFile(
+                        jobInfo,
+                        TABLE4,
+                        "cdc1.jsonl",
+                        "DataStreamToSpannerSimpleIT/mysql-cdc-AllDatatypeColumns.jsonl"),
+                    SpannerRowsCheck.builder(spannerResourceManager, TABLE4)
+                        .setMinRows(1)
+                        .setMaxRows(1)
+                        .build()))
+            .build();
+
+    // Wait for conditions
+    PipelineOperator.Result result =
+        pipelineOperator()
+            .waitForCondition(createConfig(jobInfo, Duration.ofMinutes(8)), conditionCheck);
+
+    // Assert Conditions
+    assertThatResult(result).meetsConditions();
+
+    assertAllDatatypeColumnsTableCdcContents();
+  }
+
+  @Test
+  public void migrationTestWithAllDatatypeMappingsWithInsertsOnly() {
+    // Construct a ChainedConditionCheck with 4 stages.
+    // 1. Send initial wave of events
+    // 2. Wait on Spanner to have events
+    ChainedConditionCheck conditionCheck =
+        ChainedConditionCheck.builder(
+                List.of(
+                    uploadDataStreamFile(
+                        jobInfo,
+                        TABLE5,
```

Comment:
Same comment for the next two test cases.

Reply:
done